pkg/kyverno/test/command.go

```diff
@@ -0,0 +1,536 @@
+			}	
+		}
+	} else {
+		path := filepath.Clean(dirPath[0])
+		fileDesc, err := os.Stat(path)
+		if err != nil {
+			errors = append(errors, err)
+		}
+		if fileDesc.IsDir() {
+			files, err := ioutil.ReadDir(path)
+			if err != nil {
+				errors = append(errors, fmt.Errorf("failed to read %v: %v", path, err.Error()))
+			}
+			for _, file := range files {
+				fmt.Printf("\napplying  test on file  %s...", file.Name())
+
+				yamlFile, err := ioutil.ReadFile(filepath.Join(path, file.Name()))
+				if err != nil {
+					return  sanitizederror.NewWithError("unable to read yaml", err)
+				}
+				valuesBytes, err := yaml.ToJSON(yamlFile)
+				if err != nil {
+					return  sanitizederror.NewWithError("failed to convert json", err)
+				}
+				if err := applyPoliciesFromPath(fs, valuesBytes, valuesFile, false); err != nil {
+					return sanitizederror.NewWithError("failed to apply test command", err)
+				}	
+			}
+		}
+		if len(errors) > 0 && log.Log.V(1).Enabled() {
+			fmt.Printf("ignoring errors: \n")
+			for _, e := range errors {
+				fmt.Printf("    %v \n", e.Error())
+			}
+		}
+	}
+	return nil
+}
+func getPoliciesFromPaths(fs billy.Filesystem, dirPath []string, isGit bool) (policies []*v1.ClusterPolicy, err error) {
+	var errors []error
+	if isGit {
+		for _, pp := range dirPath {
+			filep, err := fs.Open(pp)
+			bytes, err := ioutil.ReadAll(filep)
+			if err != nil {
+				fmt.Printf("Error: failed to read file %s: %v", filep.Name(), err.Error())
+			}
+			policyBytes, err := yaml.ToJSON(bytes)
+			if err != nil {
+				fmt.Printf("failed to convert to JSON: %v", err)
+				continue
+			}
+			policiesFromFile, errFromFile := ut.GetPolicy(policyBytes)
+			if errFromFile != nil {
+				err := fmt.Errorf("failed to process : %v", errFromFile.Error())
+				errors = append(errors, err)
+				continue
+			}
+			policies = append(policies, policiesFromFile...)
+		}
+	} else {
+		if len(dirPath) > 0 && dirPath[0] == "-" {
+			if common.IsInputFromPipe() {
+				policyStr := ""
+				scanner := bufio.NewScanner(os.Stdin)
+				for scanner.Scan() {
+					policyStr = policyStr + scanner.Text() + "\n"
+				}
+				yamlBytes := []byte(policyStr)
+				policies, err = ut.GetPolicy(yamlBytes)
+				if err != nil {
+					return nil, sanitizederror.NewWithError("failed to extract the resources", err)
+				}
+			}
+		} else {
+			var errors []error
+			policies, errors = common.GetPolicies(dirPath)
+			if len(policies) == 0 {
+				if len(errors) > 0 {
+					return nil, sanitizederror.NewWithErrors("failed to read file", errors)
+				}
+				return nil, sanitizederror.New(fmt.Sprintf("no file found in paths %v", dirPath))
+			}
+			if len(errors) > 0 && log.Log.V(1).Enabled() {
+				fmt.Printf("ignoring errors: \n")
+				for _, e := range errors {
+					fmt.Printf("    %v \n", e.Error())
+				}
+			}
+		}
+	}	
+	return
+}
+
+func mutatePolices(policies []*v1.ClusterPolicy) ([]*v1.ClusterPolicy, error) {
```

Comment:
is this a duplicate? can we not reuse this method?

Reply:
Yes, added in common pkg. using for both command test/apply